core/src/main/java/org/apache/accumulo/core/data/RowRange.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.accumulo.core.data;
+
+import org.apache.hadoop.io.Text;
+
+/**
+ * This class is used to specify a range of rows.
+ */
+public class RowRange {
+  final private Text startRow;
+  final private Text endRow;
+  final private boolean startRowInclusive;
+  final private boolean endRowInclusive;
+  private final boolean infiniteStartRow;
+  private final boolean infiniteEndRow;
+
+  /**
+   * Creates a range of rows that goes from negative to positive infinity
+   */
```

Comment:
This is a common way we've expressed this, but it's technically incorrect. The range goes from an empty byte array to an infinitely large byte array of all 1 bits. We should try to find a better way to express this succinctly.

Maybe "from null to positive infinity" or "from no bytes to positive infinity" or "range that includes all possible rows".

Reply:
Addressed in 46db626